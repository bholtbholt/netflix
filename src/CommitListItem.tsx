import * as React from 'react';
import { Commit } from './github';
import { CommitIcon } from '@primer/octicons-react';

export const CommitListItem: React.FC<Commit> = ({ author, commit, html_url, sha }) => {
  const formattedDate = new Date(commit.author.date).toLocaleDateString(undefined, {
    year: 'numeric',
    month: 'short',
    day: 'numeric',
  });

  return (
    <li className="flex items-start -ml-7 mt-6">
      <CommitIcon className="flex-shrink-0 mr-3 text-green-500 bg-white" size={28} />
      <div className="flex-1">
        <a
          className="text-lg font-semibold text-indigo-800 hover:text-indigo-600 mb-1 inline-block"
          href={html_url}
          rel="nofollow"
          target="_blank"
        >
          {commit.message}
        </a>
        <ul className="flex items-center flex-wrap gap-x-4 text-gray-600">
          {author ? (
            <li className="flex items-center text-sm">
              <a
                className="flex items-center font-semibold hover:underline mr-1"
                href={author.html_url}
                rel="nofollow"
                target="_blank"
              >
                <img
                  className="rounded-full mr-1"
                  src={author.avatar_url}
                  alt={`Profile picture of ${author.login}`}
                  width="24"
                  height="24"
                />
                {author.login}
              </a>
              <span className="hidden md:inline md:mr-1">committed on</span>
              {formattedDate}
            </li>
          ) : null}
          <li className="font-mono leading-5 font-semibold tracking-wide text-2xs bg-gray-100 px-2">
            {sha.substring(0, 7)}
          </li>
        </ul>
      </div>
    </li>
  );
};
