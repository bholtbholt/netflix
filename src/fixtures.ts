import { AuthorOrCommitter, Commit, CommitDetails, Repo } from './github';

export function createRepo(repo = {}): Repo {
  const mockRepo: Repo = {
    id: 8759133,
    node_id: 'MDEwOlJlcG9zaXRvcnk4NzU5MTMz',
    name: 'zuul',
    full_name: 'Netflix/zuul',
    private: false,
    owner: {
      login: 'Netflix',
      id: 913567,
      node_id: 'MDEyOk9yZ2FuaXphdGlvbjkxMzU2Nw==',
      avatar_url: 'https://avatars.githubusercontent.com/u/913567?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/Netflix',
      html_url: 'https://github.com/Netflix',
      followers_url: 'https://api.github.com/users/Netflix/followers',
      following_url: 'https://api.github.com/users/Netflix/following{/other_user}',
      gists_url: 'https://api.github.com/users/Netflix/gists{/gist_id}',
      starred_url: 'https://api.github.com/users/Netflix/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/Netflix/subscriptions',
      organizations_url: 'https://api.github.com/users/Netflix/orgs',
      repos_url: 'https://api.github.com/users/Netflix/repos',
      events_url: 'https://api.github.com/users/Netflix/events{/privacy}',
      received_events_url: 'https://api.github.com/users/Netflix/received_events',
      type: 'Organization',
      site_admin: false,
    },
    html_url: 'https://github.com/Netflix/zuul',
    description:
      'Zuul is a gateway service that provides dynamic routing, monitoring, resiliency, security, and more.',
    fork: false,
    url: 'https://api.github.com/repos/Netflix/zuul',
    forks_url: 'https://api.github.com/repos/Netflix/zuul/forks',
    keys_url: 'https://api.github.com/repos/Netflix/zuul/keys{/key_id}',
    collaborators_url: 'https://api.github.com/repos/Netflix/zuul/collaborators{/collaborator}',
    teams_url: 'https://api.github.com/repos/Netflix/zuul/teams',
    hooks_url: 'https://api.github.com/repos/Netflix/zuul/hooks',
    issue_events_url: 'https://api.github.com/repos/Netflix/zuul/issues/events{/number}',
    events_url: 'https://api.github.com/repos/Netflix/zuul/events',
    assignees_url: 'https://api.github.com/repos/Netflix/zuul/assignees{/user}',
    branches_url: 'https://api.github.com/repos/Netflix/zuul/branches{/branch}',
    tags_url: 'https://api.github.com/repos/Netflix/zuul/tags',
    blobs_url: 'https://api.github.com/repos/Netflix/zuul/git/blobs{/sha}',
    git_tags_url: 'https://api.github.com/repos/Netflix/zuul/git/tags{/sha}',
    git_refs_url: 'https://api.github.com/repos/Netflix/zuul/git/refs{/sha}',
    trees_url: 'https://api.github.com/repos/Netflix/zuul/git/trees{/sha}',
    statuses_url: 'https://api.github.com/repos/Netflix/zuul/statuses/{sha}',
    languages_url: 'https://api.github.com/repos/Netflix/zuul/languages',
    stargazers_url: 'https://api.github.com/repos/Netflix/zuul/stargazers',
    contributors_url: 'https://api.github.com/repos/Netflix/zuul/contributors',
    subscribers_url: 'https://api.github.com/repos/Netflix/zuul/subscribers',
    subscription_url: 'https://api.github.com/repos/Netflix/zuul/subscription',
    commits_url: 'https://api.github.com/repos/Netflix/zuul/commits{/sha}',
    git_commits_url: 'https://api.github.com/repos/Netflix/zuul/git/commits{/sha}',
    comments_url: 'https://api.github.com/repos/Netflix/zuul/comments{/number}',
    issue_comment_url: 'https://api.github.com/repos/Netflix/zuul/issues/comments{/number}',
    contents_url: 'https://api.github.com/repos/Netflix/zuul/contents/{+path}',
    compare_url: 'https://api.github.com/repos/Netflix/zuul/compare/{base}...{head}',
    merges_url: 'https://api.github.com/repos/Netflix/zuul/merges',
    archive_url: 'https://api.github.com/repos/Netflix/zuul/{archive_format}{/ref}',
    downloads_url: 'https://api.github.com/repos/Netflix/zuul/downloads',
    issues_url: 'https://api.github.com/repos/Netflix/zuul/issues{/number}',
    pulls_url: 'https://api.github.com/repos/Netflix/zuul/pulls{/number}',
    milestones_url: 'https://api.github.com/repos/Netflix/zuul/milestones{/number}',
    notifications_url:
      'https://api.github.com/repos/Netflix/zuul/notifications{?since,all,participating}',
    labels_url: 'https://api.github.com/repos/Netflix/zuul/labels{/name}',
    releases_url: 'https://api.github.com/repos/Netflix/zuul/releases{/id}',
    deployments_url: 'https://api.github.com/repos/Netflix/zuul/deployments',
    created_at: '2013-03-13T18:53:57Z',
    updated_at: '2021-09-11T16:32:45Z',
    pushed_at: '2021-09-01T18:55:22Z',
    git_url: 'git://github.com/Netflix/zuul.git',
    ssh_url: 'git@github.com:Netflix/zuul.git',
    clone_url: 'https://github.com/Netflix/zuul.git',
    svn_url: 'https://github.com/Netflix/zuul',
    homepage: '',
    size: 4224,
    stargazers_count: 11271,
    watchers_count: 11271,
    language: 'Java',
    has_issues: true,
    has_projects: true,
    has_downloads: true,
    has_wiki: true,
    has_pages: true,
    forks_count: 2110,
    mirror_url: null,
    archived: false,
    disabled: false,
    open_issues_count: 251,
    license: {
      key: 'apache-2.0',
      name: 'Apache License 2.0',
      spdx_id: 'Apache-2.0',
      url: 'https://api.github.com/licenses/apache-2.0',
      node_id: 'MDc6TGljZW5zZTI=',
    },
    forks: 2110,
    open_issues: 251,
    watchers: 11271,
    default_branch: 'master',
    permissions: {
      admin: false,
      push: false,
      pull: true,
    },
  };

  return { ...mockRepo, ...repo };
}

export function createCommit(commit = {}): Commit {
  const mockCommitDetails: CommitDetails = {
    author: {
      name: 'Carl Mastrangelo',
      email: 'carl-mastrangelo@users.noreply.github.com',
      date: '2021-09-01T18:55:22Z',
    },
    committer: {
      name: 'GitHub',
      email: 'noreply@github.com',
      date: '2021-09-01T18:55:22Z',
    },
    message: 'core: include proxy protocol in server conn dimms',
    tree: {
      sha: '138126c9ea71224b827889df4abbab5fd148bb48',
      url: 'https://api.github.com/repos/Netflix/zuul/git/trees/138126c9ea71224b827889df4abbab5fd148bb48',
    },
    url: 'https://api.github.com/repos/Netflix/zuul/git/commits/6d29e74fd27e6e3ad8b7efa8de511a567654cf63',
    comment_count: 0,
    verification: {
      verified: true,
      reason: 'valid',
    },
  };

  const mockAuthor: AuthorOrCommitter = {
    login: 'carl-mastrangelo',
    id: 8943572,
    node_id: 'MDQ6VXNlcjg5NDM1NzI=',
    avatar_url: 'https://avatars.githubusercontent.com/u/8943572?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/carl-mastrangelo',
    html_url: 'https://github.com/carl-mastrangelo',
    followers_url: 'https://api.github.com/users/carl-mastrangelo/followers',
    following_url: 'https://api.github.com/users/carl-mastrangelo/following{/other_user}',
    gists_url: 'https://api.github.com/users/carl-mastrangelo/gists{/gist_id}',
    starred_url: 'https://api.github.com/users/carl-mastrangelo/starred{/owner}{/repo}',
    subscriptions_url: 'https://api.github.com/users/carl-mastrangelo/subscriptions',
    organizations_url: 'https://api.github.com/users/carl-mastrangelo/orgs',
    repos_url: 'https://api.github.com/users/carl-mastrangelo/repos',
    events_url: 'https://api.github.com/users/carl-mastrangelo/events{/privacy}',
    received_events_url: 'https://api.github.com/users/carl-mastrangelo/received_events',
    type: 'User',
    site_admin: false,
  };
  const mockCommit: Commit = {
    sha: '6d29e74fd27e6e3ad8b7efa8de511a567654cf63',
    node_id: 'MDY6Q29tbWl0ODc1OTEzMzo2ZDI5ZTc0ZmQyN2U2ZTNhZDhiN2VmYThkZTUxMWE1Njc2NTRjZjYz',
    commit: mockCommitDetails,
    url: 'https://api.github.com/repos/Netflix/zuul/commits/6d29e74fd27e6e3ad8b7efa8de511a567654cf63',
    html_url: 'https://github.com/Netflix/zuul/commit/6d29e74fd27e6e3ad8b7efa8de511a567654cf63',
    comments_url:
      'https://api.github.com/repos/Netflix/zuul/commits/6d29e74fd27e6e3ad8b7efa8de511a567654cf63/comments',
    author: mockAuthor,
    committer: mockAuthor,
  };

  return { ...mockCommit, ...commit };
}
