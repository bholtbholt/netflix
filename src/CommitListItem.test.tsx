import * as React from 'react';
import renderer from 'react-test-renderer';
import { CommitListItem } from './CommitListItem';
import { createCommit } from './fixtures';

it('should match snapshot', async () => {
  const snapshot = renderer.create(<CommitListItem {...createCommit()} />).toJSON();

  expect(snapshot).toMatchSnapshot();
});
