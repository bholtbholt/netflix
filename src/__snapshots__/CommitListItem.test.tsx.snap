// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<li
  className="flex items-start -ml-7 mt-6"
>
  <svg
    aria-hidden="true"
    className="flex-shrink-0 mr-3 text-green-500 bg-white"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path fill-rule=\\"evenodd\\" d=\\"M17.5 11.75a.75.75 0 01.75-.75h5a.75.75 0 010 1.5h-5a.75.75 0 01-.75-.75zm-17.5 0A.75.75 0 01.75 11h5a.75.75 0 010 1.5h-5a.75.75 0 01-.75-.75z\\"></path><path fill-rule=\\"evenodd\\" d=\\"M12 16.25a4.5 4.5 0 100-9 4.5 4.5 0 000 9zm0 1.5a6 6 0 100-12 6 6 0 000 12z\\"></path>",
      }
    }
    fill="currentColor"
    height={28}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 24 24"
    width={28}
  />
  <div
    className="flex-1"
  >
    <a
      className="text-lg font-semibold text-indigo-800 hover:text-indigo-600 mb-1 inline-block"
      href="https://github.com/Netflix/zuul/commit/6d29e74fd27e6e3ad8b7efa8de511a567654cf63"
      rel="nofollow"
      target="_blank"
    >
      core: include proxy protocol in server conn dimms
    </a>
    <ul
      className="flex items-center flex-wrap gap-x-4 text-gray-600"
    >
      <li
        className="flex items-center text-sm"
      >
        <a
          className="flex items-center font-semibold hover:underline mr-1"
          href="https://github.com/carl-mastrangelo"
          rel="nofollow"
          target="_blank"
        >
          <img
            alt="Profile picture of carl-mastrangelo"
            className="rounded-full mr-1"
            height="24"
            src="https://avatars.githubusercontent.com/u/8943572?v=4"
            width="24"
          />
          carl-mastrangelo
        </a>
        <span
          className="hidden md:inline md:mr-1"
        >
          committed on
        </span>
        Sep. 1, 2021
      </li>
      <li
        className="font-mono leading-5 font-semibold tracking-wide text-2xs bg-gray-100 px-2"
      >
        6d29e74
      </li>
    </ul>
  </div>
</li>
`;
